### DISCUSSION TITLE
How to add time series data in milli seconds

### DISCUSSION THREAD

**Discussion by Kevin-Bobby** [XCharts-Team/XCharts#307] [Q&A] [answered]
How do I add time series data in milliseconds, it works when I add data every second, if i add it once every half second the two data overlaps in the same place, is there anyway to update the time axis once every few milliseconds. I set xAxis.type = Axis.AxisType.Time; if i change to value axis im not able to add xaxis label it automatically adds values to it. And i cant do it as category axis. please help

**Comment by monitor1394** [comment, marked as answer]
Yes, the time axis currently only supports second-level timestamps. 
For the millisecond level, the category axis is recommended.
If you use the category axis, you can convert the time to string yourself and insert it into axis data as a category.

**Reply by Kevin-Bobby** [reply]
thank you i tried that and it worked except one thing.
This is the code snippet i am using:
if (lastIndex >= size)
{
    chart.GetSerie(0).RemoveData(0);
    chart.GetSerie(1).RemoveData(0);
}
chart.AddXAxisData(datetime.ToString());
chart.AddData(0, voltage[0], "current1");
chart.AddData(1, voltage[1], "current2");

this works except the xaxis just keeps getting bigger. the graph itself is working just like i wanted it to. I search the api but couldnt find how to remove xaxis data  can you tell me how

**Reply by monitor1394** [reply]
You can try using the API:
`chart.SetMaxCache(int maxCache); `
